spec/controllers/api/internal/two_factor_authentication/webauthn_controller_spec.rb

```diff
@@ -145,5 +169,29 @@
         expect(response.status).to eq(401)
       end
     end
+
+    context 'with a configuration that does not exist' do
+      let(:params) { { id: 0 } }
+
+      it 'responds with unsuccessful result' do
+        expect(response_body).to eq(
+          success: false,
+          error: t('errors.manage_authenticator.internal_error'),
+        )
+        expect(response.status).to eq(400)
```

Comment:
> in other apps I've considered these 404's? 🤷

That's a good point. This feels partly like a limitation of the `FormResponse` pattern we're [using](https://github.com/18F/identity-idp/blob/bc9da12f2cbe63c26410159390ba2d4498e50d0c/app/controllers/api/internal/two_factor_authentication/webauthn_controller.rb#L25), since there's not a built-in way for an [added error](https://api.rubyonrails.org/v5.0/classes/ActiveModel/Errors.html#method-i-add) to signal that it should trigger a specific HTTP status code.

A couple options could be:

- Validate this in the controller, similar to [what we're already doing with the fallback no-JavaScript controller](https://github.com/18F/identity-idp/blob/bc9da12f2cbe63c26410159390ba2d4498e50d0c/app/controllers/users/webauthn_controller.rb#L8)
- Invent a custom `options` value which represents the expected status code, e.g. `errors.add(..., http_status: 404)`

Reply:
its probably not worth all that for this PR

Reply:
Sure, I'll merge this as-is, but I'd be motivated to try to find a solution to the problem described above, since it's likely we'd continue to run into it if we build more JSON API controllers.